custom_components/nest_protect/__init__.py

```diff
@@ -37,6 +37,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         nest = await client.authenticate(access_token)
     except Exception as exception:  # pylint: disable=broad-except
         LOGGER.exception(exception)
+        raise ConfigEntryNotReady from exception
```

Comment:
Good one :-). Eventually we will need to have some better exceptions, but for now this is a good one.

Reply:
Yes, catching Exception here for example isn't really the "right" as a more narrow exception would be preferred. I thought adding some tests can be a good way to get started evolving this.  I may add some in pynest next that exercise the API responses, then it might make it easier to add the subscriber stuff with more confidence

Reply:
Sounds great. pynest is still very limited indeed and doesn't handle any error handling (yet). (the reason why it is still embedded, instead of hosted on pypi).